operator/pkg/controlplane/controlplane.go

```diff
@@ -24,33 +25,49 @@ func EnsureControlPlaneComponent(component, name, namespace string, client clien
 
 	deployment, ok := deployments[component]
 	if !ok {
-		return fmt.Errorf("no exist manifest for %s", component)
+		klog.V(2).InfoS("skip applying karmada component", "component", component, "karmada", klog.KRef(namespace, name))
+		return nil
 	}
+
 	if err := apiclient.CreateOrUpdateDeployment(client, deployment); err != nil {
 		return fmt.Errorf("failed to create deployment resource for component %s, err: %w", component, err)
 	}
 	return nil
 }
 
 func getComponentManifests(name, namespace string, cfg *operatorv1alpha1.KarmadaComponents) (map[string]*appsv1.Deployment, error) {
-	kubeControllerManager, err := getKubeControllerManagerManifest(name, namespace, cfg.KubeControllerManager)
-	if err != nil {
-		return nil, err
+	manifest := map[string]*appsv1.Deployment{}
+
+	if cfg.KubeControllerManager != nil {
```

Comment:
When is cfg.KubeControllerManager eq to nil?

Reply:
Currently, kubeControllerManager  can not be nil, but the KarmadaDescheduler could be nil,  here we'd better check if them is nil.

Reply:
https://github.com/karmada-io/karmada/blob/master/operator/pkg/init.go#L225
We set defaults to karmada cr here, the mandatory components can not be nil.  I thinks add a check in func `getComponentManifests` is more rigorous.

Reply:
I feel like it's a bit overprotective and it will make people who look at the code think that this may be equal to nil.

Reply:
ok